utils/rand/rand_test.go

```diff
@@ -0,0 +1,258 @@
+package rand
+
+import (
+	"fmt"
+	"math"
+	mrand "math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"gonum.org/v1/gonum/stat"
+)
+
+// Simple unit tests using a very basic randomness test.
+// It doesn't evaluate randomness of the output and doesn't perform advanced statistical tests.
+func TestRandomIntegers(t *testing.T) {
+
+	t.Run("basic randomness", func(t *testing.T) {
+		sampleSize := 80000
+		tolerance := 0.05
+		n := 10 + mrand.Intn(100)
+		distribution := make([]float64, n)
+
+		t.Run("Uint", func(t *testing.T) {
+			// partition all outputs into `n` classes and compute the distribution
+			// over the partition. Each class has a width of `classWidth`
+			classWidth := math.MaxUint / uint(n)
+			// populate the distribution
+			for i := 0; i < sampleSize; i++ {
+				r, err := Uint()
+				require.NoError(t, err)
+				distribution[r/classWidth] += 1.0
+			}
+			stdev := stat.StdDev(distribution, nil)
+			mean := stat.Mean(distribution, nil)
+			assert.Greater(t, tolerance*mean, stdev, fmt.Sprintf("basic randomness test failed. stdev %v, mean %v", stdev, mean))
```

Comment:
similar to my comment in the other pr.  stddev is meaningless in uniform distribution.  may just check for `expectedMean * (1 - tolerance) < mean < expectedMean * (1 + tolerance)`.

Also, maybe refactor the test cases to use the same skeleton (the only change that change is how bucket width/index is computed)

Reply:
I answered your comment here: https://github.com/onflow/flow-go/pull/4067#discussion_r1143914972

Good idea about the refactoring the test common logic.

Reply:
refactor and clarifying comment about the standard dev test: [7cf858a](https://github.com/onflow/flow-go/pull/4062/commits/7cf858a36b44dde599177d9d04dc128cd8c07a33) and [003882a](https://github.com/onflow/flow-go/pull/4062/commits/003882a5913eb65a5a8a1e44f273282afef88503)